app/src/main/java/org/mozilla/focus/utils/UrlUtils.kt

```diff
@@ -161,6 +161,7 @@ object UrlUtils {
     fun toUrlBarDisplay(url: String): String {
         return when {
             url == Urls.APP_HOME -> "" // Empty string forces the EditText to show hint text
+            url.startsWith(Urls.DATA_PREFIX) -> ""
```

Comment:
Actually, we can't do this for security reason. Imagine the following attack vector:
- User goes to malicious site
- Malicious site sends user to `data:...` site that is google login page
- User sees "Search or enter address", much less suspicious than a `data:` URI

Options:
- Components can use the `loadDataWithBaseUrl` method instead of `loadData`: this would let us set our own URL (in [this code](https://github.com/mozilla-mobile/android-components/blob/14fdb2d78a009bc0dc782cb28898a6f5d3128c23/components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt#L71))
- We could whitelist our exact error page though I'd prefer to fix it the right way ^

Reply:
I filed https://github.com/mozilla-mobile/android-components/issues/865 against components.